frontend/pkg/frontend/ocm.go

```diff
@@ -470,6 +474,13 @@ func (f *Frontend) BuildCSNodePool(ctx context.Context, nodePool *api.HCPOpenShi
 			Value(float64(*nodePool.Properties.NodeDrainTimeoutMinutes)))
 	}
 
+	nodepoolVersion := nodePool.Properties.Version.ID
+	nodepoolCSversion, err := ocm.ConvertOpenshiftVersionAddPrefix(nodepoolVersion)
+	if err != nil {
+		return nil, err
+	}
+	nodePool.Properties.Version.ID = nodepoolCSversion
```

Comment:
This is happening too late; the version is already set in `npBuilder` by this point.
This code needs to happen before the `arohcpv1alpha1.NewVersion()` call above.